django/db/models/query.py

```diff
@@ -767,7 +768,7 @@ def datetimes(self, field_name, kind, order='ASC', tzinfo=None):
         else:
             tzinfo = None
         return self.annotate(
-            datetimefield=DateTime(field_name, kind, tzinfo),
+            datetimefield=Trunc(field_name, kind, output_field=DateTimeField(), tzinfo=tzinfo),
```

Comment:
Hmm nope, forgot to check that. Guess I have to put the logic back in, but it'll be added as a private implementation detail to support `queryset.datetimes()`.